raiden/api/python.py

```diff
@@ -274,6 +273,17 @@ def validate_token_app(self, token_app):
         if time_elapsed > 30:
             raise TokenAppExpired("Token app expired")
 
+    def register_secret_light(self,
+             signed_tx: typing.SignedTransaction,
+             message_id: typing.MessageID,
+             message_order: int,
+             payment_id: typing.PaymentID,
+    ):
+        message_exists = LightClientMessageHandler.is_light_client_protocol_message_already_stored_message_id(message_id, payment_id, message_order, self.raiden.wal)
+        if not message_exists:
+            raise InvalidPaymentIdentifier()
+        self.raiden.default_secret_registry.register_secret_light(signed_tx)
```

Comment:
I think we are ignoring the result of this onchain operation. 

What happens if the TX fails? we update the message on the database anyway?

Reply:
If the transaction fails, an exception is thrown (and catched in raiden.py). So it will never reach the code that updates the message.